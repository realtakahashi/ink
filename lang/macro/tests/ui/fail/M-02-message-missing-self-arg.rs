use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod missing_message_self_arg {
    #[ink(storage)]
    struct MissingMessageSelfArg {}

    impl MissingMessage {
        #[ink(constructor)]
        fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        fn missing_self_arg() {}
    }
}

fn main() {}
