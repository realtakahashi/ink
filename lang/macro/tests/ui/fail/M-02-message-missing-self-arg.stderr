error: ink! messages and methods must have a `self` receiver
 --> $DIR/M-02-message-missing-self-arg.rs:3:1
  |
3 | #[ink::contract(version = "0.1.0")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
